registry.go

```diff
@@ -0,0 +1,53 @@
+package narwhal
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/heroku/docker-registry-client/registry"
+	log "github.com/sirupsen/logrus"
+)
+
+// ListTagsByRepo uses the OCI v2 interface to query a registry for the tags associated
+// with a registry.
+func ListTagsByRepo(ctx context.Context, registryURL, repo, username, password string) ([]string, error) {
```

Comment:
When I see a function that takes more than two string arguments, I prefer to extract a struct so that the caller can label which input is which. The risk is that the caller may accidentally mess up the order of the arguments.

Reply:
I like the more than 2 argument rule.  I went back and forth on adding a struct.